plasma/root_chain/contracts/RootChain/RootChain.sol

```diff
@@ -193,15 +195,30 @@ contract RootChain {
         exit memory currentExit = exits[exitsQueue.getMin()];
         uint256 blknum = currentExit.utxoPos.div(1000000000);
         while (childChain[blknum].created_at < twoWeekOldTimestamp && exitsQueue.currentSize() > 0) {
-            currentExit.owner.transfer(currentExit.amount);
+            // currentExit.owner.transfer(currentExit.amount);
+            // Prevents attacker from submitting an address of a malicous contract that could revert transfer()
+            balances[currentExit.owner] = balances[currentExit.owner].add(currentExit.amount);
             uint256 priority = exitsQueue.delMin();
             delete exits[priority];
             delete exitIds[currentExit.utxoPos];
             currentExit = exits[exitsQueue.getMin()];
         }
     }
 
-    /* 
+    // @dev Allows any to withdraw their funds from the contract (eg. exit refunds, bonds, etc.)
+    // @dev Isolates transfers so attacker can't halt essential processes by reverting transfer()
+    function withdraw()
+        public
+        returns (uint256)
+    {
+        require(balances[msg.sender] > 0);
+        uint256 transferAmount = balances[msg.sender];
+        delete balances[msg.sender];
+        msg.sender.transfer(transferAmount);
+        return transferAmount;
```

Comment:
Might be worth emitting a `Withdrawal(address, uint256)` event instead of returning `transferAmount` here.